legal-api/src/legal_api/resources/v2/business/colin_sync.py

```diff
@@ -60,43 +61,54 @@ def get_completed_filings_for_colin():
-                else:
-                    filing_json['filing']['business'] = business.json()
-            elif not filing_json['filing']['business'].get('legalName'):
-                filing_json['filing']['business']['legalName'] = business.legal_name
-
-            if filing.filing_type == 'correction' and business.legal_type != Business.LegalTypes.COOP.value:
+
+        filing_json = copy.deepcopy(filing.filing_json)
+        filing_json['filingId'] = filing.id
+        filing_json['filing']['header']['source'] = Filing.Source.LEAR.value
+        filing_json['filing']['header']['date'] = filing.filing_date.isoformat()
+        filing_json['filing']['header']['learEffectiveDate'] = filing.effective_date.isoformat()
+
+        if not filing_json['filing'].get('business'):
+            if filing.transaction_id:
+                business_revision = VersionedBusinessDetailsService.get_business_revision_obj(
+                    filing.transaction_id, business.id)
+                filing_json['filing']['business'] = VersionedBusinessDetailsService.business_revision_json(
+                    business_revision, business.json())
+            else:
+                # should never happen unless its a test data created directly in db.
+                # found some filing in DEV, adding this check to avoid exception
+                filing_json['filing']['business'] = business.json()
+        elif not filing_json['filing']['business'].get('legalName'):
+            filing_json['filing']['business']['legalName'] = business.legal_name
+
+        if filing.filing_type == 'correction' and business.legal_type != Business.LegalTypes.COOP.value:
+            try:
                 set_correction_flags(filing_json, filing)
-            elif (filing.filing_type == 'amalgamationApplication' and
-                    filing_json['filing']['amalgamationApplication']['type'] in [
-                        Amalgamation.AmalgamationTypes.horizontal.name,
-                        Amalgamation.AmalgamationTypes.vertical.name]):
-                try:
-                    set_from_primary_or_holding_business_data(filing_json, filing)
-                except Exception as ex:  # noqa: B902
-                    current_app.logger.info(ex)
-                    continue  # do not break this function because of one filing
-            elif (filing.filing_type == 'dissolution' and
-                  filing.filing_sub_type == 'involuntary'):
-                batch_processings = BatchProcessing.find_by(filing_id=filing.id)
-                if not batch_processings:
-                    continue  # skip filing for missing batch processing info
+            except Exception as ex:  # noqa: B902
+                current_app.logger.error(f'correction: filingId={filing.id}, error: {str(ex)}')
+                # to skip this filing and block subsequent filing from syncing in update-colin-filings
+                filing_json['filing']['header']['name'] = None
```

Comment:
This will help us to skip in sync (https://github.com/bcgov/lear/blob/main/jobs/update-colin-filings/update_colin_filings.py#L83) and block any subsequent filings for that business